src/main/com/example005/radek/printDiagonalsApplication.java

```diff
@@ -0,0 +1,57 @@
+package main.com.example005.radek;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+enum DiagonalType {
+    LEFT,
+    RIGHT,
+    BOTH
+}
+
+public class printDiagonalsApplication {
+
+    public static void main(String[] args) {
+
+        int size = getEnteredValue();
+        printDiagonals(DiagonalType.LEFT, size);
+        printDiagonals(DiagonalType.RIGHT, size);
+        printDiagonals(DiagonalType.BOTH, size);
+    }
+
+    static int getEnteredValue() {
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter size of diagonal: ");
+        try {
+            return scanner.nextInt();
+        } catch (InputMismatchException nfe) {
+            System.out.println("You have to insert numeric value!");
+            System.exit(0);
+            return 0;
+        }
+    }
+
+    static void printDiagonals(DiagonalType type, int size) {
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j <= size; j++) {
+                switch (type) {
```

Comment:
pro kazdy print novou metodu ... printLeft() ...

Reply:
Hotovo

Reply:
tak a ted muzes pouzit enhanced switch a bude to krasne citelne ... lambda zapis... tady to je lehke a ocividne na pochopeni, ne lambda jak jindy, ktera pracuje s kolekcemi

Reply:
Hotovo